### PR TITLE
pallet-timestamp: `UnixTime::now` implementation logs error only if called at genesis

### PR DESCRIPTION
**Pull request by gpestana**
This PR reverts the removal of an [`if` statement](https://github.com/paritytech/polkadot-sdk/commit/7ecf3f757a5d6f622309cea7f788e8a547a5dce8#diff-8bf31ba8d9ebd6377983fd7ecc7f4e41cb1478a600db1a15a578d1ae0e8ed435L370) merged recently, which affected test   output verbosity of several pallets (e.g. staking, EPM, and potentially others).

More generally, the `UnixTime::now` implementation of the timestamp pallet should log an error *only* when called at the genesis block.

### CODE REVIEW

**Comment by bkchr** [comment]
> This PR reduces the unnecessary verbosity in non-std environments when calling `UnixTime::now()` implemented by the timestamp pallet.

About which environments are you talking here about?

**Comment by gpestana** [comment]
> About which environments are you talking here about?

`feature = test`, for example.

**Comment by bkchr** [comment]
> `feature = test`, for example.

Where is this run? In CI?

**Comment by gpestana** [comment]
> > `feature = test`, for example.
> 
> Where is this run? In CI?

Yep, and locally too. For example, if you run the test of the pallet-staking in master locally, you'll see the error message logged from the timestamp pallet due to removing the `sp_std::if_std!` check. Now, the tests won't fail but the error log is output and annoying.

File: substrate/frame/timestamp/src/lib.rs [hunk 1]
```diff
@@ -368,6 +368,7 @@ impl<T: Config> UnixTime for Pallet<T> {
 		// `sp_timestamp::InherentDataProvider`.
 		let now = Now::<T>::get();
 
+		#[cfg(not(feature = "std"))]
```
Comment (bkchr) [inline substrate/frame/timestamp/src/lib.rs:371]:
```
			if now == T::Moment::zero() {
             log::error!(
			target: "runtime::timestamp",
			"`pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0",
		);
}
```

This is the proper fix. The linked pr omitted the if..

Comment (gpestana) [reply substrate/frame/timestamp/src/lib.rs:371]: Added the if statement back, thanks.

**Review by ggwpez — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by ggwpez** [comment]
FYI `R0-silent` means that this change will not trigger a release of this crate.

**Comment by bkchr** [comment]
Then @gpestana please add some `prdoc`.

**Review by acatangiu — APPROVED** [review]

**Review by seadanda — APPROVED** [review]

**Pull request merged**